### DISCUSSION TITLE
[@types/ember] @ember/runloop debounce type does not match docs

### DISCUSSION THREAD

**Discussion by jacobq** [DefinitelyTyped/DefinitelyTyped#60206] [Issues with a @types package]
According to the documentation, [`debounce`](https://api.emberjs.com/ember/4.3/functions/@ember%2Frunloop/debounce) can take 3, 4 or 5 arguments (even back in [v2.x](https://api.emberjs.com/ember/2.18/functions/@ember%2Frunloop/debounce)), but this doesn't work:

```ts
// Should be OK but causes TS2554: Expected 3-4 arguments, but got 5.
debounce(this, this.someFunc, { /* optional arg */ }, 100, true);
```

Dropping the optional argument makes the error go away, but it looks like the type definitions are not properly recognizing the last two arguments (`wait` & `immediate`):
![image](https://user-images.githubusercontent.com/1633459/166973448-1e2ff828-681a-44d5-a30b-94083c9eaffd.png)
I'm guessing the problem is coming from what the "ts-expect-error" is ignoring here:
https://github.com/emberjs/ember.js/blob/21b3733485ad364b5ac71c3423ab821b6e3ded4a/packages/%40ember/runloop/index.ts#L862-L865

Looks like the typedefs I'm getting in my IDE are:
```ts
export function debounce<Target, M extends EmberMethod<Target>>(
    ...args: [
        target: Target,
        method: M,
        ...args: EmberMethodParams<Target, M>,
        wait: number,
        immediate?: boolean
    ]
): EmberRunTimer;
```

However, I admittedly don't know how to read that so am going to try to just ignore this type checking error for now.

[a comment by an automated account is not shown]

**Comment by chriskrycho** [comment]
@jacobq sorry, missed this notification until just now. The *intent* is to support the variadic arguments by way of that spread, but it seems not to be working. Can’t promise to put this at an *especially* high priority, but I’ll note it as one of the things to revisit as we look to pull the types into Ember proper in the next couple months!
